tribe-ext-template-outliner.php

```diff
@@ -89,45 +104,44 @@ private function get_settings() {
 
 		/**
 		 * Extension initialization and hooks.
+		 *
+		 * @since 1.0.0
 		 */
 		public function init() {
-			// Load plugin textdomain
-			// Don't forget to generate the 'languages/tribe-ext-template-outliner.pot' file
+			// Load plugin textdomain.
 			load_plugin_textdomain( 'tribe-ext-template-outliner', false, basename( dirname( __FILE__ ) ) . '/languages/' );
 
 			if ( ! $this->php_version_check() ) {
 				return;
 			}
 
-			// Don't run in the admin or anything silly like that!
-			if (
-				is_admin()
-				|| wp_doing_ajax()
-			) {
-				return;
-			}
+			$this->class_loader();
+
+			$this->get_settings();
 
-			// Don't show to the general public - in case someone actually uses it on a live site!
+			// Only show to the roles allowed in the settings.
 			$user = wp_get_current_user();
-			if ( ! in_array( 'administrator', (array) $user->roles ) ) {
+			$allowed = $this->settings->get_option( 'roles', [] );
```

Comment:
I watched your video and like this setting and its code better than the $user->roles but why empty array by default? Is Administrator not enabled upon initial plugin install?

Reply:
Administrator is enabled by default - but what if someone disables all roles (as a way to "disable" it)? Then passing a default value would circumvent that option.

Reply:
I've changed it now so admin is the default and the checkboxes cannot **all** be empty.